### PR TITLE
[Bug] fix bugs of analyzation of rewrited SQL when triggered the materialized vew

### LINKED ISSUE #9876: [Bug] Unexpected Exception of unknown column when query statement having table alias triggered the matarailized view
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.0

### What's Wrong?

Execution of a query statement having a `COUNT` aggregate function which could trigger a materialized view will get the unknown column exception, and the column name corresponds to the rewrited materialized column of the parameter of aggregate function.

### What You Expected?

return the result set normally

### How to Reproduce?

1. 
```sql
CREATE TABLE case_tbl (col1 varchar, col2 varchar, col3 int)
DISTRIBUTED BY HASH(col3)
BUCKETS 3
PROPERTIES(
    "replication_num"="1"
);
```

2. 
```sql
create materialized view case_mv as select col2, count(col1) as c from d_test2 d_test2 group by col1, col2;
```

3.
```sql
select col2, count(col1) as c from d_test2 d_test2  t group by col1, col2;
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: #9876

## Problem Summary:

Remove the redundant logic of set tableName for the new create slotRef based on the rewrited materialized view column which could cause the unexpected analyze exception

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/mvrewrite/CountFieldToSum.java [hunk 1]
```diff
@@ -81,15 +80,12 @@ public Expr apply(Expr expr, Analyzer analyzer, ExprRewriter.ClauseType clauseTy
...
     }
 
-    private Expr rewriteExpr(SlotRef queryColumnSlotRef, Column mvColumn, Analyzer analyzer) {
+    private Expr rewriteExpr(Column mvColumn, Analyzer analyzer) {
         Preconditions.checkNotNull(mvColumn);
-        Preconditions.checkNotNull(queryColumnSlotRef);
-        TableName tableName = queryColumnSlotRef.getTableName();
-        Preconditions.checkNotNull(tableName);
-        SlotRef mvSlotRef = new SlotRef(tableName, mvColumn.getName());
+        SlotRef mvSlotRef = new SlotRef(null, mvColumn.getName());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/rewrite/mvrewrite/CountFieldToSum.java:88]: Comment

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/rewrite/mvrewrite/CountFieldToSum.java:94]: done

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — COMMENTED** [review]
Please add test case for this

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**